### ISSUE TITLE
Review how elements are highlighted

### ISSUE THREAD

**Issue by carolinerosin** [usnistgov/hl7-diff-tool-frontend#13]
This is something we need to bring up with Rob/the workgroup before making any change, because I'm not sure if there was a reason why it was done this way ?

I think we should only highlight elements based on the comparison that we are looking at. For example here
![image](https://user-images.githubusercontent.com/13591511/166790826-72bdbb1e-c66f-4b70-bbff-6613bee1bbaa.png)

I am looking at "cardinality". But I see a lot of element highlighted because they have changes in other attributes (maybe usage or datatype for example).
When I pick "cardinality" I'm only expected elements with different cardinalities to be highlighted, hopefully that makes sense ?

**Comment by mellouli-dev** [comment]
@carolinerosin Yes, I agree with you. 
Note that those highlighted elements might contain children where a cardinality was changed. So even if we highlight elements based on the selected attribute (usage, cardinality etc...), some elements might still be highlighted.

**Comment by carolinerosin** [comment]
yep, I agree, if an element has a change, the element and all its parents should be highlighted.

So for example if a field has a usage change, when I'm selecting usage, I expect to see the field, the segment that contains that field (and group(s) containing the segment, if applicable) highlighted. Then if I change to "cardinality", and that segment does not contain an element with a cardinality difference, then the field and segment are not highlighted.
